webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/users/UserManagementService.scala

```diff
@@ -86,19 +101,22 @@ object UserManagementIO {
    * - only one can be executed in parallel,
    * - in case of error, a back-up (the original file) is written back.
    */
-  def transformUserFile(xmlFile: File, transformer: Elem => PureResult[RewriteRule]): IOResult[Unit] = {
-    semaphore.withPermit(for {
-      readable   <- IOResult.attempt(xmlFile.isReadable)
-      _          <- ZIO.when(!readable)(Unexpected(s"${xmlFile.path} is not readable").fail)
-      writable   <- IOResult.attempt(xmlFile.isWritable)
-      _          <- ZIO.when(!writable)(Unexpected(s"${xmlFile.path} is not writable").fail)
-      parsedFile <- IOResult.attempt(ConstructingParser.fromFile(xmlFile.toJava, preserveWS = true))
-      userXML    <- IOResult.attempt(parsedFile.document().children)
-      toReplace  <- replaceXml(userXML, transformer, xmlFile.pathAsString).toIO
-      p           = new RudderPrettyPrinter(300, 4)
-      xmlString   = p.formatNodes(toReplace)
-      _          <- withBackup(xmlFile)(IOResult.attempt(xmlFile.overwrite(xmlString)))
-    } yield ())
+  @nowarn("msg=a type was inferred to be `Any`; this may indicate a programming error.")
+  def transformUserFile(xmlFile: File, transformer: Elem => PureResult[RewriteRule]): ZIO[Scope, RudderError, Unit] = {
+    ZIO
+      .service[Scope]
+      .flatMap(_.extend(for {
```

Comment:
Same as above: tells that we extends the scope in environment with that method without closing the scope at the end of the method.